framework/src/binary_manager/binary_manager_update.c

```diff
@@ -30,31 +30,30 @@
 #include <binary_manager/binary_manager.h>
 #include "binary_manager_internal.h"
 
-int binary_manager_update_binary(char *binary_name)
+binmgr_result_type_e binary_manager_update_binary(char *binary_name)
 {
 	int ret;
 	binmgr_request_t request_msg;
 
 	if (binary_name == NULL || strlen(binary_name) > BIN_NAME_MAX - 1) {
 		bmdbg("load_new_binary failed : invalid binary name\n");
-		return BINMGR_INVALID_PARAM;
+		return ret;
 	}
 
 	ret = binary_manager_set_request(&request_msg, BINMGR_UPDATE, binary_name);
 	if (ret < 0) {
-		return BINMGR_COMMUNICATION_FAIL;
+		return ret;
```

Comment:
This work actually, but to make it more explicitly, 'binmgr_result_type_e ret' is better option.